gravitee-apim-gateway/gravitee-apim-gateway-policy/src/main/java/io/gravitee/gateway/jupiter/policy/DefaultPolicyFactory.java

```diff
@@ -34,10 +37,16 @@ public class DefaultPolicyFactory implements PolicyFactory {
     private final ConcurrentMap<String, Policy> policies = new ConcurrentHashMap<>();
     private final PolicyPluginFactory policyPluginFactory;
     private final io.gravitee.gateway.policy.PolicyFactory v3PolicyFactory;
+    private final ExpressionLanguageConditionFilter<ConditionalPolicy> filter;
 
-    public DefaultPolicyFactory(final PolicyPluginFactory policyPluginFactory, final ConditionEvaluator<String> conditionEvaluator) {
+    public DefaultPolicyFactory(
+        final PolicyPluginFactory policyPluginFactory,
+        final ExpressionLanguageConditionFilter<ConditionalPolicy> filter
+    ) {
         this.policyPluginFactory = policyPluginFactory;
-        v3PolicyFactory = new io.gravitee.gateway.policy.impl.PolicyFactoryImpl(policyPluginFactory, conditionEvaluator);
+        this.filter = filter;
+        // V3 policy factory doesn't need condition evaluator anymore as condition is directly handled by jupiter.
+        this.v3PolicyFactory = new io.gravitee.gateway.policy.impl.PolicyFactoryImpl(policyPluginFactory, null);
```

Comment:
does it worth a new constructor without a null param ?

Reply:
Yes you're right, I'll add another constructor 👍